### PR TITLE
Feat/request payment cards 16737

### LINKED ISSUE #16737: Request Payment - chat cards
Implement mini card in status chat input and card displayed in chat.

### PR DESCRIPTION
**Pull request by Cuteivist**
Closes #16737 

### What does the PR do

* Implemented request payment cards displayed in chat input and chat history
* Added additional status messages to storybook
* Added unit tests

### Affected areas

Chat

### Architecture compliance

- [X] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [X] I've checked the design and this PR matches it

image

image

### CODE REVIEW

File: storybook/pages/StatusMessagePage.qml [hunk 1]
```diff
@@ -30,6 +32,7 @@ SplitView {
                 isAReply: false
                 trustIndicator: StatusContactVerificationIcons.TrustedType.Verified
                 outgoingStatus: StatusMessage.OutgoingStatus.Delivered
+                requeestPaymentModel: []
```
Comment (caybro) [inline storybook/pages/StatusMessagePage.qml:35]:
```suggestion
                requestPaymentModel: []
```

File: storybook/qmlTests/tests/tst_StatusMessage.qml [hunk 2]
```diff
@@ -35,7 +35,7 @@ Item {
     property StatusMessage controlUnderTest: null
 
     TestCase {
-        name: "TokenSelectorView"
+        name: "StatusMessage"
```
Comment (caybro) [inline storybook/qmlTests/tests/tst_StatusMessage.qml:38]: :+1:

File: storybook/stubs/shared/stores/RequestPaymentStore.qml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import QtQuick 2.15
+
+QtObject {
```
Comment (caybro) [inline storybook/stubs/shared/stores/RequestPaymentStore.qml:3]: Should be (ideally) empty...

File: ui/imports/shared/controls/chat/RequestPaymentMiniCardDelegate.qml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                    font.pixelSize: Theme.additionalTextSize
+                    font.weight: Font.Medium
+                }
+                StatusBaseText {
+                    Layout.fillWidth: true
+                    Layout.fillHeight: true
+                    font.pixelSize: Theme.tertiaryTextFontSize
+                    color: Theme.palette.baseColor1
+                    verticalAlignment: Text.AlignVCenter
+                    text: "%1 %2".arg(root.amount).arg(root.symbol)
```
Comment (caybro) [inline ui/imports/shared/controls/chat/RequestPaymentMiniCardDelegate.qml:80]: The amount should be formatted?

File: ui/imports/shared/views/chat/LinksMessageView.qml [hunk 5]
```diff
@@ -25,15 +25,21 @@ Flow {
...
+    property string senderName
+    property var senderImageAssetSettings
+
     readonly property alias hoveredLink: linksRepeater.hoveredUrl
     property string highlightLink: ""
 
     signal imageClicked(var image, var mouse, string imageSource, string url)
     signal openContextMenu(var item, string url, string domain)
     signal setNeverAskAboutUnfurlingAgain(bool neverAskAgain)
+    signal requestPaymentClicked(var symbol, var amount, var address, var chainId)
```
Comment (caybro) [inline ui/imports/shared/views/chat/LinksMessageView.qml:42]:
```suggestion
    signal requestPaymentClicked(string symbol, string amount, string address, int chainId)
```

sth like that? surely not all should be `var`

File: ui/imports/shared/views/chat/LinksMessageView.qml [hunk 6]
```diff
@@ -56,6 +62,21 @@ Flow {
         sourceComponent: enableLinkComponent
     }
 
+    Repeater {
+        id: requestPaymentRepeater
+        model: root.requestPaymentModel
+        delegate: RequestPaymentCardDelegate {
+            required property var model
+            objectName: "RrequestPaymentDelegate_" + model.index
```
Comment (caybro) [inline ui/imports/shared/views/chat/LinksMessageView.qml:70]:
```suggestion
            objectName: "requestPaymentDelegate_" + model.index
```

File: ui/imports/shared/controls/chat/ChatInputLinksPreviewArea.qml [hunk 7]
```diff
@@ -96,6 +104,17 @@ Control {
                     onImageRemoved: root.imageRemoved(index)
                     visible: !!imagePreviewArray && imagePreviewArray.length > 0
                 }
+                Repeater {
+                    id: requestPaymentRepeater
```
Comment (caybro) [inline ui/imports/shared/controls/chat/ChatInputLinksPreviewArea.qml:108]: `id` needed?

File: ui/imports/shared/controls/chat/ChatInputLinksPreviewArea.qml [hunk 8]
```diff
@@ -32,15 +32,23 @@ Control {
...
     readonly property alias hoveredUrl: d.hoveredUrl
-    readonly property bool hasContent: imagePreviewArray.length > 0 || showLinkPreviewSettings || linkPreviewRepeater.count > 0
+    readonly property bool hasContent: imagePreviewArray.length > 0 || showLinkPreviewSettings || linkPreviewRepeater.count > 0 || paymentRequestRepeater.count > 0
 
     signal imageRemoved(int index)
     signal imageClicked(var chatImage)
     signal linkReload(string link)
     signal linkClicked(string link)
 
+    signal removePaymentRequestPreviewRequested(int index)
```
Comment (caybro) [inline ui/imports/shared/controls/chat/ChatInputLinksPreviewArea.qml:50]: Ugh, what a mouthful :laughing:

File: ui/imports/shared/controls/chat/PaymentRequestMiniCardDelegate.qml [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+                    Layout.fillWidth: true
+                    Layout.fillHeight: true
+                    StatusBaseText {
+                        Layout.maximumWidth: parent.width * 0.8
+                        Layout.fillHeight: true
+                        font.pixelSize: Theme.tertiaryTextFontSize
+                        color: Theme.palette.baseColor1
+                        verticalAlignment: Text.AlignVCenter
+                        elide: Text.ElideRight
+                        text: root.amount
```
Comment (caybro) [inline ui/imports/shared/controls/chat/PaymentRequestMiniCardDelegate.qml:84]: Must format this value!

File: ui/imports/shared/controls/delegates/PaymentRequestCardDelegate.qml [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+    padding: 12
+
+    Behavior on borderColor {
+        ColorAnimation { duration: 200 }
+    }
+
+    QtObject {
+        id: d
+        property real bannerImageMargins: 1 / Screen.devicePixelRatio // image size isn't pixel perfect..
+        property bool highlight: root.highlight || root.hovered
```
Comment (caybro) [inline ui/imports/shared/controls/delegates/PaymentRequestCardDelegate.qml:46]: `readonly` those two?

File: ui/imports/shared/controls/delegates/PaymentRequestCardDelegate.qml [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+        Item {
+            Layout.fillWidth: true
+            Layout.preferredHeight: 4
+        }
+
+        StatusBaseText {
+            Layout.fillWidth: true
+            wrapMode: Text.WrapAtWordBoundaryOrAnywhere
+            text: qsTr("Send %1 %2 to %3").arg(root.amount).arg(root.symbol).arg(Utils.compactAddress(root.address.toLowerCase(), 4))
```
Comment (caybro) [inline ui/imports/shared/controls/delegates/PaymentRequestCardDelegate.qml:121]: Format the `amount`

File: ui/imports/shared/views/chat/MessageView.qml [hunk 12]
```diff
@@ -986,6 +992,13 @@ Loader {
                         gifUnfurlingEnabled: root.sharedRootStore.gifUnfurlingEnabled
                         canAskToUnfurlGifs: !root.sharedRootStore.neverAskAboutUnfurlingAgain
                         onSetNeverAskAboutUnfurlingAgain: root.sharedRootStore.setNeverAskAboutUnfurlingAgain(neverAskAgain)
+                        onPaymentRequestClicked: (index) => {
+                            const receiver = StatusQUtils.ModelUtils.get(paymentRequestModel, index, "receiver")
+                            const amount = StatusQUtils.ModelUtils.get(paymentRequestModel, index, "amount")
+                            const symbol = StatusQUtils.ModelUtils.get(paymentRequestModel, index, "symbol")
+                            const chainId = StatusQUtils.ModelUtils.get(paymentRequestModel, index, "chainId")
```
Comment (caybro) [inline ui/imports/shared/views/chat/MessageView.qml:999]: You could get the whole row at once, and then get the props

**Review by caybro — COMMENTED** [review]
LGTM in general, just format the `amount` pls when displaying it

**Review by caybro — APPROVED** [review]
LGTM

File: ui/imports/shared/popups/send/SendModal.qml
Comment (alexjba) [inline ui/imports/shared/popups/send/SendModal.qml:1]: Do we need a regression on the SendModal before merging?

**Review by alexjba — APPROVED** [review]
LGTM! Nice work 👍

File: ui/imports/shared/popups/send/SendModal.qml
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/popups/send/SendModal.qml:1]: I think @alexjba point is to test any regressions on the SendModal before merging right ?

**Pull request merged**